File: LivrusVirtus/wxWindows/wxRuby/CadPaises.rb
require 'wxruby'
require 'Cadastros'
require 'TextValidatorCtrl'
require 'ConsultasSQL'
require 'resource'
require 'RotinasGlobais'

class FrmCadPaises < FrmCadastros
  def initialize(parent)
    super(parent, TextValidatorCtrl::FILTER_ALPHA, true) 
    @edtCodigo.set_max_length(3)
    @labDescricao.set_label('País:')

    evt_button(ID_FRMCADASTROS_BTNNOVO) {|event| OnBtnNovoClick(event)}
    evt_button(ID_FRMCADASTROS_BTNPESQUISAR) {|event| OnBtnPesquisarClick(event)}
    evt_button(ID_FRMCADASTROS_BTNANTERIOR) {|event| OnBtnAnteriorClick(event)}
    evt_button(ID_FRMCADASTROS_BTNPROXIMO) {|event| OnBtnProximoClick(event)}
    evt_button(ID_FRMCADASTROS_BTNSALVAR) {|event| OnBtnSalvarClick(event)}
    evt_button(ID_FRMCADASTROS_BTNEXCLUIR) {|event| OnBtnExcluirClick(event)}
    evt_button(ID_FRMCADASTROS_BTNLIMPAR) {|event| OnBtnLimparClick(event)}
    evt_button(ID_FRMCADASTROS_BTNFECHAR) {|event| OnBtnFecharClick(event)}
  end

  def ValidaDados(bTodosDados)
    if $Rotinas.ValidaCampo(
      @edtCodigo.get_value(), Resource::STR_CODNINF) then
      return false
    end
    if bTodosDados then
  	  if $Rotinas.ValidaCampo(
        @edtDescricao.get_value(), Resource::STR_PAISNINF) then
    		return false
      end
    end
    return true
  end

  def OnBtnNovoClick(event)
    self.btnNovoClick()
    self.NovoDado($ConsSQL.Pais('N', @edtCodigo.get_value(),""))
    event.skip()
  end

  def OnBtnPesquisarClick(event)
    self.PesquisarDados($ConsSQL.Pais('S', 
        @edtCodigo.get_value(),""), Resource::STR_PAISNENC)
    event.skip()
  end

  def OnBtnAnteriorClick(event)
    self.registroAnterior()
    event.skip()
  end

  def OnBtnProximoClick(event)
    self.registroProximo()
    event.skip()
  end

  def OnBtnSalvarClick(event)
    self.SalvarDados(self.ValidaDados(true),
      $ConsSQL.Pais('S', @edtCodigo.get_value(),""),
      $ConsSQL.Pais('U', @edtCodigo.get_value(), @edtDescricao.get_value()),
      $ConsSQL.Pais('I', @edtCodigo.get_value(), 
      @edtDescricao.get_value()), true)
    event.skip()
  end

  def OnBtnExcluirClick(event)
    self.ExcluirDados(self.ValidaDados(false),
      $ConsSQL.Pais('D', @edtCodigo.get_value(),""))
    event.skip()
  end

  def OnBtnLimparClick(event)
    self.btnLimparClick()
    event.skip()
  end

  def OnBtnFecharClick(event)
    self.get_parent().close(true)
    event.skip()
  end
end

File: LivrusVirtus/BoaConstructor/wxPython/CadPaises.py
#Boa:FramePanel:frmCadPaises

from wxPython.wx import *
from TextValidator import TextValidator
from Cadastros import frmCadastros
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
from resource import Resource

def create(parent):
    return frmCadPaises(parent)

class frmCadPaises(frmCadastros):
    def _init_utils(self):
        pass

    def _init_ctrls(self, prnt):
        frmCadastros._init_ctrls(self, prnt, TextValidator.FILTER_ALPHA, True)
        self._init_utils()
        self.edtCodigo.SetMaxLength(3)
        self.labDescricao.SetLabel(u'Pa\xeds:')
        EVT_BUTTON(self.btnNovo, self.btnNovo.GetId(), self.OnBtnNovoClick)
        EVT_BUTTON(self.btnFechar, self.btnFechar.GetId(), 
          self.OnBtnFecharClick)
        EVT_BUTTON(self.btnPesquisar, self.btnPesquisar.GetId(),
              self.OnBtnPesquisarClick)
        EVT_BUTTON(self.btnAnterior, self.btnAnterior.GetId(),
              self.OnBtnAnteriorClick)
        EVT_BUTTON(self.btnProximo, self.btnProximo.GetId(),
              self.OnBtnProximoClick)
        EVT_BUTTON(self.btnSalvar, self.btnSalvar.GetId(),
              self.OnBtnSalvarClick)
        EVT_BUTTON(self.btnExcluir, self.btnExcluir.GetId(),
              self.OnBtnExcluirClick)
        EVT_BUTTON(self.btnLimpar, self.btnLimpar.GetId(),
              self.OnBtnLimparClick)
                
    def __init__(self, parent):
        self._init_ctrls(parent)
        
    def ValidaDados(self, bTodosDados):
      if RotinasGlobais.Rotinas.ValidaCampo(
        self.edtCodigo.GetValue(), Resource.STR_CODNINF):
        return False
      if bTodosDados:
    	  if RotinasGlobais.Rotinas.ValidaCampo(
          self.edtDescricao.GetValue(), Resource.STR_PAISNINF):
      		return False;
      return True

    def OnBtnFecharClick(self, event):
      self.GetParent().Close(True)
    
    def OnBtnNovoClick(self, event):
      self.btnNovoClick()
      event.Skip()

    def OnBtnPesquisarClick(self, event):
      self.PesquisarDados(ConsultasSQL.ConsSQL.Pais('S', 
        self.edtCodigo.GetValue(),""), Resource.STR_PAISNENC)
      event.Skip()

    def OnBtnAnteriorClick(self, event):
      self.registroAnterior()
      event.Skip()

    def OnBtnProximoClick(self, event):
      self.registroProximo()
      event.Skip()

    def OnBtnSalvarClick(self, event):
      self.SalvarDados(self.ValidaDados(True),
        ConsultasSQL.ConsSQL.Pais('S', self.edtCodigo.GetValue(),""),
        ConsultasSQL.ConsSQL.Pais('U', self.edtCodigo.GetValue(), 
        self.edtDescricao.GetValue()),
        ConsultasSQL.ConsSQL.Pais('I', self.edtCodigo.GetValue(), 
        self.edtDescricao.GetValue()), True)
      event.Skip()

    def OnBtnExcluirClick(self, event):
      self.ExcluirDados(self.ValidaDados(False),
        ConsultasSQL.ConsSQL.Pais('D', self.edtCodigo.GetValue(),""))
      event.Skip()

    def OnBtnLimparClick(self, event):
      self.btnLimparClick()
      event.Skip()

